python/web/src/web.py

```
@@ -922,6 +923,40 @@ def create_file():
     if not process["status"]:
         return response(error=True, message=process["msg"])
 
+    # Formatting and injecting driver, if one is choosen
+    if drive_format:
+        driver_base_path = Path(f"{WEB_DIR}/../../../HFSer/drivers")
+        volume_name = f"HD {size / 1024 / 1024:0.0f}M"
+        known_drivers = [
+                "Apple Drive Setup 1.7.3",
+                "Apple Drive Setup 2.0.3",
+                "Apple HD SC Setup 7.3.5",
+                "FWB 4.5.2",
+                "Lido 7.56",
+                "MacOS8 Installer Driver",
+                ]
+        if drive_format in known_drivers:
```

You can avoid a level of indentation by failing early on this, e.g.

```
if drive_format not in known_drivers:
    response()

other stuff goes here
```

This brings the error handling close to the actual check in the code, which is nice for readability.

I'm sure you don't need me to tell you this! 😝

---

Good call! For some reason I always avoid negative checks whenever possible, but in this case the negative check is definitely better semantically.